### PR TITLE
Table refactoring

### PR DESCRIPTION
**Pull request by OlegKuzovkov**

### CODE REVIEW

File: src/web/data_types/__init__.py [hunk 1]
```diff
@@ -54,55 +54,60 @@
...
+    def __init__(self, value):
+        super().__init__(By.CLASS_NAME, value)
 
 
 class CssSelector(Locator):
 
-    def __new__(cls, value):
-        return By.CSS_SELECTOR, value
+    def __init__(self, value):
+        super().__init__(By.CSS_SELECTOR, value)
```
Comment (ovaillancourt) [inline src/web/data_types/__init__.py:113]: If you want to get rid of a lot of the repetition in this file you can use the 3-params form of the `type()` function to create your classes dynamically (and maybe have a small utility function to make it even shorter).

```python
# Utility function
def genLocatorType(name, by)
    return type(name, (Locator), dict(by=by))

# Call it to define each class
ClassName   = genLocatorType('ClassName', By.CLASS_NAME)
CssSelector = genLocatorType('CssSelector', By.CSS_SELECTOR)
# ...
```

reference on the `type(,,)` function if needed -> https://docs.python.org/3/library/functions.html#type

File: src/web/data_types/table.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            elif out or time.time() > end_time:
+                break
+        return out
+
+
+class Charts(Table):
+    """
+        Prefix it with chr_
+    """
+    pass
```
Comment (ovaillancourt) [inline src/web/data_types/table.py:142]: What is the value of having this class (`Charts`) if it's just a type definition?

Also - if this refers to actual `charts` from our dashboard I think this is too specialized to go in KOM Framework and should actually go in the actual test suite in the Thirdshelf code base (ie: the framework could be used in a lot of contexts where there are no charts so having a `Charts` type would be spurious).

The same comment might also apply to the `Menu` type defined in another `element_list_types.py` depending on its intended purpose.

**Review by ovaillancourt — COMMENTED** [review]
Good stuff - see the few comments but overall nothing significant to address.

File: src/web/data_types/table.py (hunk 2 above)
Comment (OlegKuzovkov) [reply src/web/data_types/table.py:142]: I am trying to reflect the actual type of the elements I am working with on a page_object level. Even if those classes are empty for now, they might have some specific functionality in the future. I know I have a lot of those, but this is the tradeoff I am making now to save visibility and have a place to extend its functionality in the future.

File: src/web/data_types/__init__.py (hunk 1 above)
Comment (OlegKuzovkov) [reply src/web/data_types/__init__.py:113]: The goal here is to have a short Locator type definition in the page_object level:
`ClassName(<value>)`.  `genLocatorType(<value> , ClassName)` - looks ugly :)

**Pull request merged**